bond-core porting to ros2

kishornaik10 opened the pull request:
bond-core porting to ros2

kishornaik10 committed:
bond-core porting to ros2

kishornaik10 committed:
added linter test cases.

kishornaik10 committed:
modified travis.yml file.

kishornaik10 committed:
Readme added in test_bond, Copyright information modified

kishornaik10 committed:
ros2_migration_readme.md is added, it contains overview, test procedure, changes related to migration.

kishornaik10 committed:
ros2_migration_readme.md is added and modified, it contains overview, build and test procedure, and changes related to migration.

kishornaik10 committed:
ros2_migration_readme.md is modified, it contains overview, build and test procedure, and changes related to migration.

kishornaik10 committed:
ros2_migration_readme.md is modified, ssh link added

kishornaik10 committed:
ros2_migration_readme.md file modified, errors removed

kishornaik10 committed:
travis.yml and ros2_migration_readme.md files modified

mjcarroll commented on bond/package.xml:
```diff
@@ -1,31 +1,34 @@
+<package format="3">
   <name>bond</name>
-  <version>1.8.3</version>
+  <version>1.8.4</version>
```
This is probably worth a major version bump.

mjcarroll replied:
To 2.0.0

kishornaik10 replied:
@mjcarroll 
I have changed the version to 2.0.0

mjcarroll commented on bond/package.xml:
```diff
@@ -1,31 +1,34 @@
   <maintainer email="michael@openrobotics.org">Michael Carroll</maintainer>
-  <license>BSD</license>
-
+  <license>Apache License 2.0</license>
```
Since the original repo was BSD, we need to stay with that in absence of approval of the original authors.

kishornaik10 replied:
@mjcarroll 
I have restored the license information, I have kept previous license and added new one but kept as commented, If needed it can be removed.

mjcarroll commented on bond/CMakeLists.txt:
```diff
@@ -1,14 +1,33 @@
+rosidl_generate_interfaces(${PROJECT_NAME}
+  "msg/Constants.msg"
+  "msg/Status.msg"
+  DEPENDENCIES std_msgs builtin_interfaces
```
Also `ADD_LINTER_TESTS`

kishornaik10 replied:
@mjcarroll , Linter tests are already added in this CMakeList

mjcarroll commented on bond/package.xml:
```diff
@@ -1,31 +1,34 @@
-  <build_depend>std_msgs</build_depend>
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <buildtool_depend>std_msgs</buildtool_depend>
+  <buildtool_depend>builtin_interfaces</buildtool_depend>
```
`std_msgs` and `builtin_interfaces` should be `build` and `exec` depends.

mjcarroll replied:
And not `buildtool_depend`

kishornaik10 replied:
@mjcarroll I have made changes as per suggestion

mjcarroll commented on bond_core/package.xml:
```diff
@@ -2,29 +2,29 @@
 <package format="2">
   <name>bond_core</name>
-  <version>1.8.3</version>
+  <version>1.8.4</version>
```
Bump to 2.0

kishornaik10 replied:
Changed the version to 2.0.0

mjcarroll commented on bond_core/package.xml:
```diff
@@ -2,29 +2,29 @@
   <maintainer email="michael@openrobotics.org">Michael Carroll</maintainer>
-  <license>BSD</license>
-
+  <license>Apache License 2.0</license>
```
Same license comment.

kishornaik10 replied:
Restored the  license information

mjcarroll reviewed (changes requested):
I'm not sure where all the diffs are coming from on the changelogs, but it doesn't seem like they should be there.  Did you regenerate the changelogs in this case?

mjcarroll commented on bondcpp/CMakeLists.txt:
```diff
@@ -1,41 +1,77 @@
+  ${rcl_LIBRARIES}
+  ${rcutils_LIBRARIES})
+
+include_directories(${INCLUDE_DIRS})
```
A bunch of these ${INCLUDE_DIRS} and ${LIBS} can be eliminated by using `ament_target_dependencies`

kishornaik10 replied:
@mjcarroll 
Not sure how to use,
can you give example?

mjcarroll commented on bondcpp/include/bondcpp/bond.hpp:
```diff
@@ -1,3 +1,17 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
```
Don't change the license.

kishornaik10 replied:
I have restored the license information, I have kept previous license and added new one but kept as commented, If needed it can be removed.

kishornaik10 replied:
And also added bondpy with a COLCON_IGNORE

mjcarroll commented on bondcpp/package.xml:
```diff
@@ -2,26 +2,47 @@
 <package format="2">
   <name>bondcpp</name>
-  <version>1.8.3</version>
+  <version>1.8.4</version>
```
version

kishornaik10 replied:
Changed it to 2.0.0

mjcarroll commented on bondcpp/package.xml:
```diff
@@ -2,26 +2,47 @@
   </description>
   <maintainer email="michael@openrobotics.org">Michael Carroll</maintainer>
-  <license>BSD</license>
+  <license>Apache License 2.0</license>
```
license

kishornaik10 replied:
Restored license information

mjcarroll commented on bondcpp/CMakeLists.txt:
```diff
@@ -1,41 +1,77 @@
+
+find_package(ament_cmake REQUIRED)
+find_package(ament_cmake_ros REQUIRED)
 find_package(Boost REQUIRED)
```
Do we have a continued dependency on Boost?

kishornaik10 replied:
@mjcarroll  We have removed dependency on Boost.

mjcarroll reviewed (changes requested):
I think it would also make more sense to retain bondpy, but mark it with a `COLCON_IGNORE` or `AMENT_IGNORE` file.

kishornaik10 committed:
CHANGELOG.rst file restored for each subpackage

kishornaik10 committed:
Added bondpy, bondpy is not migrated in ROS2 yet

kishornaik10 committed:
bondcpp/CMakeLists.txt modified, removed dependency on Boost as it is not used in code.

kishornaik10 committed:
test_bond_srv_gen removed, now Testbond.srv is genrated in test_bond

kishornaik10 committed:
package.xml file is modified for bond and test_bond, std_msgs and builtin_interfaces made to build and exec depends.

kishornaik10 committed:
Version in each subpackage is changed to 2.0.0.

kishornaik10 committed:
license information restored, enabled BSD, disabled Apache License 2.0

kishornaik10 commented:
@mjcarroll  Thank you for your feedback, changes have been done and added with different commits.
I have restored the license information, I have kept previous license and added new one but kept as commented, If needed it can be removed.

kishornaik10 committed:
statemap.py file added, modifications done for linter test.

kishornaik10 committed:
Renamed smclibcpp to smclib

kishornaik10 commented:
Hello @mjcarroll 
I have restored smclib, as earlier I had separated it as  smclibcpp and smclibpy.
I have addressed all your suggestions.
Please give feedback.
Thanks,

norro commented:
Hello @kishornaik10, thanks for the PR. Is it correct that bondpy migration has not started, yet? Are there any existing plans on doing so?

mjcarroll commented:
Closing in favor of #54
